include/knowhere/sparse_utils.h

```diff
@@ -59,6 +60,31 @@ GetDocValueBM25Computer(float k1, float b, float avgdl) {
     };
 }
 
+class DocIdFilterByVector {
+ public:
+    DocIdFilterByVector(std::vector<table_t>&& docids) : docids_(std::move(docids)) {
+        std::sort(docids_.begin(), docids_.end());
+    }
+
+    [[nodiscard]] bool
+    test(const table_t id) const {
+        // find the first id that is greater than or equal to the specific id
+        while (pos_ < docids_.size() && docids_[pos_] < id) {
+            ++pos_;
+        }
+        return !(pos_ < docids_.size() && docids_[pos_] == id);
+    }
+
+    [[nodiscard]] bool
+    empty() const {
+        return docids_.empty();
+    }
+
+ private:
+    std::vector<table_t> docids_;
+    mutable size_t pos_ = 0;
```

Comment:
this is a relatively untypical solution. The more traditional solution would be make `pos_` non-mutable and remove constness from `test()` in order to remove the confusion. Maybe, even change `test()` name, because it is not really `test()`. Or one may remove `const`-ness in a `Cursor` (see a comment below)

Would you please elaborate the reason for such an unusual implementation?